src/types.ts

```diff
@@ -35,5 +35,32 @@ export interface PRDetails {
   title: string
   description: string
 }
+interface File {
+  source: string,
+  destination: string
+}
+
+export interface TemplateConfig {
+  version: string,
+  automerge: boolean,
+  files: Array<File>,
+  values: object
+}
+
+export interface CreateCheckInput {
+  owner: string,
+  repo: string,
+  sha: string
+}
+
+export interface UpdateCheckInput {
+  owner: string,
+  repo: string,
+  sha: string,
+  result: string,
+  check_run_id: number
+}
```

Comment:
We could merge these into a single `Check` type:

```suggestion
export interface Check {
  owner: string,
  repo: string,
  sha: string,
  result?: string,
  check_run_id?: number
}
```

Reply:
I'd rather extend it, the `result` and `check_run_id` are not optional on the UpdateCheckInput

Reply:
Ah, you could do this then: 

```typescript
export interface CheckInput {
  owner: string,
  repo: string,
  sha: string,
}

export interface CheckOutput extends CheckInput {
  result?: string,
  check_run_id?: number
}
```

or via types (whichever you prefer):

```typescript
export type CheckInput {
  owner: string,
  repo: string,
  sha: string,
}

export type CheckOutput & CheckInput {
  result?: string,
  check_run_id?: number
}
```